src/chocolatey.resources/helpers/functions/Get-ChocolateyUnzip.ps1

```diff
@@ -196,6 +197,31 @@ param(
     default { throw "7-Zip signalled an unknown error (code $exitCode)" }
   }
 
-  $env:ChocolateyPackageInstallLocation = $destination
-  return $destination
+  if ((test-path variable:url) -AND (@($url).count -gt 0))
+  {
+    # If our unzip resulted in a .tar file that matches our input zip file without it's extensions, unzip it
+    # does not work with archive names that contain a '.' before the extensions
+    $matchingtarname = "$destinationNoRedirection\$((split-path -leaf $url).split('.')[0]).tar"

```

Comment:
I intentionally kept the scope very narrow to the most common use case of a matching tar name inside another enclosing archive.

If the functionality were to unzip non-matching names it would seem to need to look for and unzip all archives if there are more than one as it is not clear which is the target.

This would also seem to cause it to unzip a bunch of sub-archives in conditions where it is not necessary - for instance archives that have sub-archives for an internal install routine to process if and when certain features are selected.

If it did unzip any archives it found - it's not clear to me how that might affect additional chocolatey procedures that work on the folder.

So I don't really have a broad enough understanding of the rest of the procedures this folder will be subject to, to go broader than the common use case I mentioned above - obviously someone working on the product for a long time could make those judgements.